src/Analyzers/CSharp/Tests/RemoveUnusedMembers/RemoveUnusedMembersTests.cs

```diff
@@ -1518,6 +1554,22 @@ public async Task IndexerIsIncrementedAndValueDropped()
             await VerifyCS.VerifyCodeFixAsync(code, code);
         }
 
+        [WorkItem(43191, "https://github.com/dotnet/roslyn/issues/43191")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task IndexerIsIncrementedAndValueDropped_VerifyAnalizerMessage()
+        {
+            var code = @"class MyClass
+{
+    private int this[int x] { get { return 0; } set { } }
+    public void M1(int x) => ++this[x];
+}";
+
+            await VerifyCS.VerifyAnalyzerAsync(code, new DiagnosticResult(
+                CSharpRemoveUnusedMembersDiagnosticAnalyzer.s_removeUnreadMembersRule)
+                .WithSpan(3, 17, 3, 21)
+                .WithArguments("MyClass.this[]"));
+        }
```

Comment:
need sibling test showing this same code, but something additional that then shows we don't emit this.